### ISSUE TITLE
The `autoNumeric:formatted` event should be triggered when the input field is cleared while continuously pressing the `Backspace` or `Delete` keys

### ISSUE THREAD

**Issue by mikekiska** [autoNumeric/autoNumeric#621]
### Current behavior
When the input field has a value and the caret is positioned at the end of the input field and the backspace key is held down to clear the value, the event autoNumeric:formatted is not triggered.

### Expected behavior
The autoNumeric:formatted event should be triggered under the conditions described above since the formatted value has changed.

### Steps to reproduce the problem
1. Use autoNumeric version 4.5.1
2. In the browser Chrome 71.0.3578.98
3. On the Windows 10 operating system
4. Then instantiate the autoNumeric object with the following options:
```js
new AutoNumeric(domElement);
```

### Link to live example:  [Codepen Example](https://codepen.io/anon/pen/xmYLYJ)
In the Codepen example you can backspace out one number at a time and the formatted event will be displayed along with the formatted value. The formatted event is triggered for every change except for when the formatted value changes to empty. If you press and hold the backspace key down until the field is cleared you will see that the formatted event is never triggered.

**Comment by AlexandreBonneau** [comment]
Nice catch there @mikekiska!
Thanks for the live example too *(..that I took the liberty of [modifying](https://codepen.io/AnotherLinuxUser/pen/Neyoaw?editors=1010) a bit)*.

**Comment by AlexandreBonneau** [comment]
Ok, that was a fun problem to search and solve.

First, when you continuously press the `Backspace` (or `Delete` key for that matter), your browser output multiple `keydown`/`input` events, and only one `keyup` event.
In the code, I checked if the target value on `keyup` was similar to the initial value saved during `keydown`, in order to check if there was effectively a `change` there.
However, since the `keydown` handler was called multiple times, specially when the user have finished clearing the input, then I would compare `''` to...`''`, and therefore no `change` was detected.
You can check that if you paste `10,000,000,000,000.00` in the input, then press `Backspace` for a few ms, then the `autoNumeric:formatted` event is correctly sent, as long as the resulting value is not empty.

So, in order to fix that bug, I needed to introduce a way to *see the past* by checking if the current `keydown` event has been preceded by a `keyup` event. Only then, I was able to check for multiple `keydown` events in a row:
```js
if ((multipleKeydownEvents && targetValue !== this.initialRawValueOnFirstKeydown) ||
            targetValue !== this.initialValueOnKeydown) {
```

**Comment by mikekiska** [comment]
@AlexandreBonneau Should the autoNumeric:formatted event also be triggered when the resulting value is empty after holding Backspace to clear the field, since the formatted value has changed?

**Comment by AlexandreBonneau** [comment]
It should be sent indeed if the field was not empty already when you started holding the Backspace key. Has the fix missed some use case?

**Comment by mikekiska** [comment]
There are two cases when the autoNumeric:formatted event is still not being triggered:
1. When the field has an initial value and the the backspace key is held down to clear it.
2. When a value is entered into the field and the backspace key is pressed once per character. In this case the event is not triggered when the formatted value becomes an empty string.

**Comment by mikekiska** [comment]
Yes, I can still reproduce it using Chrome and Firefox with [this Codepen](https://codepen.io/anon/pen/xmYLYJ). If you load the Codepen, set the caret position to the end of the field and then hold backspace to clear the field it does not trigger the autoNumeric:formatted event.

**Comment by AlexandreBonneau** [comment]
This is fixed (again, for real this time!) in `v4.5.7`.

Please test, just to be sure ;P
